### PR TITLE
Add flag --fixed-form-infer

### PR DESCRIPTION
**Pull request by meow464**
At the moment we only use the file extension .f, in the future as problems arise we can add more heuristics like what values appear on the first 5 columns.

~~Is there any reason this has not been done yet?~~

~~Ok, six tests fail, if those are the only reasons I'm wiling to work on them.~~

### CODE REVIEW

File: src/bin/lfortran.cpp [hunk 1]
```diff
@@ -2051,6 +2051,12 @@ int main(int argc, char *argv[])
         if (CLI::NonexistentPath(arg_file).empty())
             throw LCompilers::LCompilersException("File does not exist: " + arg_file);
 
+        // Decide if a file is fixed format based on the extension
+        // Gfortran does the same thing
+        if (arg_file.size() >= 2 && arg_file.substr(arg_file.size() - 2) == ".f") {
```
Comment (Shaikh-Ubaid) [inline src/bin/lfortran.cpp:2056]: I think we have a function `endswith()`, that we can utilize here. It is currently used at the following place:

https://github.com/lfortran/lfortran/blob/3c7beebd85502e4907f98f17cf842937b586501e/src/bin/lfortran.cpp#L2170

**Comment by Shaikh-Ubaid** [comment]
It seems Dr. Ondrej shared here https://github.com/lfortran/lfortran/pull/2095#issuecomment-1665927321 that we should consider `.f` files as free modern fortran by default. I am unsure what we should do.

**Comment by meow464** [comment]
I see, CMake expects the compiler to figure out whether code is fixed or free form and gfortran uses the file extension to do that. 
We are hitting this limitation when trying to compile lapack for exemple.

**Comment by certik** [comment]
Yes, `.f` should be treated as free-form. You can use `--fixed-form` to compile fixed-form files.

CMake should be updated to pass the correct options, since it knows if it is `.f` or `.f90` and if CMake wants to assume that `.f` is fixed-form, it can do it, and it should then pass `--fixed-form` to LFortran.

**Comment by meow464** [comment]
Maybe we could have a --fixed-form-heuristics where the compiler tries to guess if a file is fixed form? We could use the file extension but also what characters appear on the first five columns.

This would be useful for existing code bases that mix fixed and free form and assume the compiler will tell them apart.

**Comment by certik** [comment]
Yes, we can definitely add `--fixed-form-heuristics`.

**Comment by meow464** [comment]
@Shaikh-Ubaid did your previous pull request include any work towards this?

**Review by certik — APPROVED** [review]
I think that looks good, thanks!

**Pull request merged**